sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/booleanExpressions.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.types.BooleanType
+
+/**
+ * String to indicate which boolean test selected.
+ */
+object BooleanTest {
+  val TRUE = "TRUE"
+  val FALSE = "FALSE"
+  val UNKNOWN = "UNKNOWN"
+
+  def calculate(input: Any, booleanValue: String): Boolean = {
+    booleanValue match {
+      case TRUE => input == true
+      case FALSE => input == false
+      case UNKNOWN => input == null
+      case _ => throw new AnalysisException("Boolean test value must be one of TRUE, " +
+        "FALSE and UNKNOWN.")
+    }
+  }
+}
```

Comment:
Shall we avoid to use `booleanValue: String`? Also, we don't need to have enum-like strings `val TRUE = "TRUE"` and so on. The following will be better in `Scala` world.
```scala
object BooleanTest {
  def calculate(input: Any, booleanValue: Option[Boolean]): Boolean = {
    booleanValue match {
      case Some(true) => input == true
      case Some(false) => input == false
      case None => input == null
      case _ => throw new AnalysisException("Boolean test value must be one of TRUE, " +
        "FALSE and UNKNOWN.")
    }
  }
}
```

Reply:
@dongjoon-hyun Thanks for your example. I get it.